File: Core.Shared/DTOs/Authorization/PermissionRuleDto.cs
using Core.Shared.Enums.Authorization;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core.Shared.DTOs.Authorization
{
    public class PermissionRuleDto
    {
        public Guid PermissionId { get; private set; }

        public RuleType Type { get; private set; }
      
        public string? FieldName { get; private set; }

        public string? JoinEntity { get; private set; }
        public string? JoinLocalKey { get; private set; }
        public string? JoinForeignKey { get; private set; }
        public string? JoinField { get; private set; }

        public ComparisonOperator Operator { get; private set; }
        public string? Value { get; private set; }

        public LogicalOperator LogicalOperator { get; private set; } 
        public int GroupOrder { get; private set; }
    }
}

File: FrontEnd/src/modules/Authorization/models/PermissionRuleDto.ts
import { LogicalOperator ,ComparisonOperator } from "./PermissionRuleEnum";

// src/modules/Authorization/models/PermissionRuleDto.ts
export interface PermissionRuleDto {
  id: string;
  permissionId: string;
  fieldName?: string;
  joinDetailId?: string;
  operator: ComparisonOperator;
  value: string;
  logicalOperator: LogicalOperator;
  groupOrder: number;

   joinLocalKey: string;
  joinForeignKey: string;
  joinEntity: string;
} 
 
